Add msvc runtime value to profile

Psy-Kai opened the pull request:
Changelog: Fix: Properly generate qbs profile for msvc.
Docs: Omit

The current qbsprofile generator does not work for msvc. `qbs config` adds architecture and target platform definitions automatically. This will be handled properly now. Also the msvc compiler.runtime values is now added to the qbs profile.

- [ ] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

Commit:
Implement

Commit:
Add missing ","

Commit:
Merge remote-tracking branch 'refs/remotes/conan/develop' into add_msvc_runtime_value_to_profile

Commit:
Merge remote-tracking branch 'refs/remotes/conan/develop' into add_msvc_runtime_value_to_profile

Commit:
Move qbs profile test into proper directory

Commit:
Handle qbs config output for msvc properly

And add tests specific for msvc.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/conan-io/conan?pullRequest=9122) <br/>All committers have signed the CLA.

Commit:
Move test for qbs build helper

Accidentally removed somehow in passed commits. So I will move it now to
the proper directory.

Commit:
Move test for qbs build helper

Accidentally removed somehow in passed commits. So I will move it now to
the proper directory.

memsharded commented:
Hi @Psy-Kai 

Thanks for contributing this. A couple of details:

- The tests in Win are failing, could you please take a look?
- It would be necessary to sign the CLA, please check the above

We plan to release 1.38 early next week, if we want this in, it would be necessary to be merge it tomorrow.

Commit:
Merge branch 'add_msvc_runtime_value_to_profile' of github.com:Psy-Kai/conan into add_msvc_runtime_value_to_profile

Commit:
Test against msvc Visual Studio version 16

Commit:
Merge remote-tracking branch 'refs/remotes/conan/develop' into
add_msvc_runtime_value_to_profile

memsharded reviewed (commented):
VS 16 is not in CI servers yet, we are testing with VS15.
In the same way, the specific compiler version might vary with respect of what you have locally installed.

Commit:
skip msvc version specific test

Commit:
Conan build server uses msvc 15

memsharded commented on conan/tools/qbs/qbsprofile.py:
```diff
@@ -220,20 +231,25 @@ class QbsProfile(object):
                 qbs.buildVariant: "{{ build_variant }}"
                 {%- endif %}
-                {%- if architecture %}
+                {%- if not _profile_values_from_setup["qbs.architecture"] and architecture %}
```
This might go also in the context definition (The context decides the "architecture" value, the template renders it)

memsharded reviewed (commented):
Looking good, check that minor comment and I think it can be merged for 1.38

Commit:
Move check if qbs.architecture is already available to context method
